client/components/share/facebook-share-preview/index.jsx

```diff
@@ -7,36 +8,60 @@
 import PropTypes from 'prop-types';
 import React, { PureComponent } from 'react';
 import { localize } from 'i18n-calypso';
+import { isNull } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import FacebookPreview from 'components/seo/facebook-preview';
 
 export class FacebookSharePreview extends PureComponent {
 	static propTypes = {
+		articleContent: PropTypes.string,
+		articleSummary: PropTypes.string,
 		articleUrl: PropTypes.string,
+		externalName: PropTypes.string,
 		externalProfilePicture: PropTypes.string,
 		externalProfileUrl: PropTypes.string,
-		externalName: PropTypes.string,
 		imageUrl: PropTypes.string,
 		message: PropTypes.string,
+		seoTitle: PropTypes.string,
+		siteIcon: PropTypes.string,
+	};
+
+	state = {
+		isProfileImageBroken: false,
 	};
 
+	setBrokenProfileImage = () => this.setState( { isProfileImageBroken: true } );
+
 	render() {
 		const {
+			articleContent,
+			articleSummary,
 			articleUrl,
+			externalDisplay,
 			externalProfilePicture,
 			externalProfileUrl,
-			externalDisplay,
 			imageUrl,
 			message,
+			seoTitle,
+			siteIcon,
 			translate,
 		} = this.props;
+		const { isProfileImageBroken } = this.state;
 		return (
 			<div className="facebook-share-preview">
 				<div className="facebook-share-preview__content">
 					<div className="facebook-share-preview__header">
 						<div className="facebook-share-preview__profile-picture-part">
-							<img
-								className="facebook-share-preview__profile-picture"
-								src={ externalProfilePicture }
-							/>
+							{ ! isProfileImageBroken && (
+								<img
+									className="facebook-share-preview__profile-picture"
+									src={ externalProfilePicture }
+									onError={ this.setBrokenProfileImage }
```

Comment:
This all feels quite... circular 🙃 Can we not determine the validity of the profile image higher up, and send down a regular prop, getting rid of the need for local state management?

Reply:
I'm not quite sure how to do what you mean?
There has been a day while I was working on this PR, that the Graph API was partially down or broken.
What happened was that we had the correct profile picture URL, but it returned `4xx`.

While I was researching the issue, I've found out that it's something quite common for the Graph API (there have been reports of similar issues for years).
Maybe "common" is not the right word. Maybe it's just "sometimes", but it's also unpredictable enough, and it really makes our preview look broken.

That `onError` is a quick check to avoid showing the broken image icon, when just the image CDN (or whatever fails) is down, but the rest of the API is up.

Reply:
We can't verify if the image is valid (I don't think valid is the right word. Maybe _available_?) until we try to load it. It may be a bit overkill, but we can take `facebook-share-preview__profile-picture` and make it a small component out of it that'll handle the `onError`, and maybe retry after a while.